File: 049-Group-Anagrams-prime.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        int[] primes = {2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103};
        Map<Integer, List<String>> map = new HashMap<>();
        for(String str:strs) {
            int k = 1;
            for(char ch:str.toCharArray()) {
                k *= primes[ch-'a'];
            }
            if(!map.containsKey(k)) {
                map.put(k, new ArrayList<>());
            }
            map.get(k).add(str);
        }
        return new ArrayList<List<String>>(map.values());
    }
}

File: 049-Group-Anagrams-prime.py
def groupAnagrams(strs):
    """
    :type strs: List[str]
    :rtype: List[List[str]]
    """
    primes = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103];  

    ans = {}
    ret = []
    for s in strs:    	
    	temp = 1
    	for j in range(len(s)):
    		print(s[j])
    		temp = temp * primes[(ord(s[j]) - ord('a'))]
    	if temp not in ans:
    		ans[temp] = []
    	ans[temp].append(s)

    for key in ans:
    	ret.append(ans[key])
    return ret

strs = ["eat", "tea", "tan", "ate", "nat", "bat"]
strs2 = ["", ""]
print(groupAnagrams(strs2))